Cython/Utility/Buffer.c

```diff
@@ -53,14 +53,7 @@ static void __Pyx_RaiseBufferFallbackError(void) {
 
 /////////////// BufferFormatStructs.proto ///////////////
 //@proto_block: utility_code_proto_before_types
-
-#ifdef __cplusplus
-template<typename T>
-struct __PYX_IS_UNSIGNED_IMPL {static const bool value = T(0) < T(-1);};
-#define __PYX_IS_UNSIGNED(type) __PYX_IS_UNSIGNED_IMPL<type>::value
-#else
-#define __PYX_IS_UNSIGNED(type) (((type) -1) > 0)
-#endif
+//@requires: Overflow.c::Common
```

Comment:
I don't think we need the `__Pyx_check_twos_complement()` part in `Common.c` for this. Thus, it might be worth extracting the unsigned checks into a new, separate utility code section and let both existing utility code sections depend on that.

Reply:
You're right, I've seen this and thought that it probably won't cause any problems, but I'm not sure about it so this seems like a better option